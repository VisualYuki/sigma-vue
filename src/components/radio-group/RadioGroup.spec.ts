import {mount, type VueWrapper} from '@vue/test-utils'
import {beforeEach, describe, expect, it} from 'vitest'

import type {RadioGroupItem} from './types'

import RadioGroup from './radio-group.vue'

describe('radio-group', () => {
	let wrapper: VueWrapper<InstanceType<typeof RadioGroup>>

	describe('when items is object', () => {
		beforeEach(async () => {
			wrapper = mount(RadioGroup, {
				props: {
					items: [
						{label: '1', value: '1'},
						{label: '2', value: '2'},
						{label: '3', value: '3'}
					] satisfies RadioGroupItem[]
				}
			})

			await wrapper.findAll('input[type="radio"]')[0]?.trigger('change')
		})

		describe('when clicking on first', () => {
			it('render snapshot', () => {
				expect(wrapper.html()).toMatchSnapshot()
			})

			it('selected first radio', () => {
				expect(wrapper.emitted('update:modelValue')?.[0]?.[0]).toBe('1')
			})

			describe('then clicking on second radio', () => {
				it('selected second radio', async () => {
					await wrapper.findAll('input[type="radio"]')[1]?.trigger('change')
					expect(wrapper.emitted('update:modelValue')?.[1]?.[0]).toBe('2')
				})

				it('render snapshot', () => {
					expect(wrapper.html()).toMatchSnapshot()
				})
			})
		})

		describe('when inizialized with particular radio', () => {
			it('correct selected', async () => {
				await wrapper.setProps({modelValue: 1})
				expect((wrapper.findAll('input[type="radio"]')[0]?.element as HTMLInputElement).checked).toBe(true)
			})
		})
	})

	describe('when items is string', () => {
		beforeEach(async () => {
			wrapper = mount(RadioGroup, {
				props: {
					items: [1, 2, 3]
				}
			})

			await wrapper.findAll('input[type="radio"]')[0]?.trigger('change')
		})

		it('render snapshot', () => {
			expect(wrapper.html()).toMatchSnapshot()
		})
	})
})
