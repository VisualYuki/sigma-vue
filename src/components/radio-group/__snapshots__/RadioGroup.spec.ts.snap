// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`radio-group > when items is object > when clicking on first > render snapshot 1`] = `
"<fieldset class="inline-flex gap-3 p-4 pe-12 relative border border-stone-500 rounded-md">
  <legend class="px-2 text-xs"></legend><label class="flex items-center select-none group leading-none flex-wrap disabled-utility"><input class="sr-only peer" type="radio" value="1">
    <div class="relative w-4 h-4 rounded-full border-2 focus-utility border-green-500"><span class="absolute top-[50%] left-[50%] translate-[-50%] w-1.5 h-1.5 rounded-full bg-green-500"></span></div>
    <p class="ms-1.5 text-stone-500">1</p>
  </label><label class="flex items-center select-none group leading-none flex-wrap disabled-utility"><input class="sr-only peer" type="radio" value="2">
    <div class="relative w-4 h-4 rounded-full border-2 focus-utility border-stone-500">
      <!--v-if-->
    </div>
    <p class="ms-1.5 text-stone-500">2</p>
  </label><label class="flex items-center select-none group leading-none flex-wrap disabled-utility"><input class="sr-only peer" type="radio" value="3">
    <div class="relative w-4 h-4 rounded-full border-2 focus-utility border-stone-500">
      <!--v-if-->
    </div>
    <p class="ms-1.5 text-stone-500">3</p>
  </label>
</fieldset>"
`;

exports[`radio-group > when items is object > when clicking on first > then clicking on second radio > render snapshot 1`] = `
"<fieldset class="inline-flex gap-3 p-4 pe-12 relative border border-stone-500 rounded-md">
  <legend class="px-2 text-xs"></legend><label class="flex items-center select-none group leading-none flex-wrap disabled-utility"><input class="sr-only peer" type="radio" value="1">
    <div class="relative w-4 h-4 rounded-full border-2 focus-utility border-green-500"><span class="absolute top-[50%] left-[50%] translate-[-50%] w-1.5 h-1.5 rounded-full bg-green-500"></span></div>
    <p class="ms-1.5 text-stone-500">1</p>
  </label><label class="flex items-center select-none group leading-none flex-wrap disabled-utility"><input class="sr-only peer" type="radio" value="2">
    <div class="relative w-4 h-4 rounded-full border-2 focus-utility border-stone-500">
      <!--v-if-->
    </div>
    <p class="ms-1.5 text-stone-500">2</p>
  </label><label class="flex items-center select-none group leading-none flex-wrap disabled-utility"><input class="sr-only peer" type="radio" value="3">
    <div class="relative w-4 h-4 rounded-full border-2 focus-utility border-stone-500">
      <!--v-if-->
    </div>
    <p class="ms-1.5 text-stone-500">3</p>
  </label>
</fieldset>"
`;

exports[`radio-group > when items is string > render snapshot 1`] = `
"<fieldset class="inline-flex gap-3 p-4 pe-12 relative border border-stone-500 rounded-md">
  <legend class="px-2 text-xs"></legend><label class="flex items-center select-none group leading-none flex-wrap disabled-utility"><input class="sr-only peer" type="radio" value="1">
    <div class="relative w-4 h-4 rounded-full border-2 focus-utility border-green-500"><span class="absolute top-[50%] left-[50%] translate-[-50%] w-1.5 h-1.5 rounded-full bg-green-500"></span></div>
    <p class="ms-1.5 text-stone-500">1</p>
  </label><label class="flex items-center select-none group leading-none flex-wrap disabled-utility"><input class="sr-only peer" type="radio" value="2">
    <div class="relative w-4 h-4 rounded-full border-2 focus-utility border-stone-500">
      <!--v-if-->
    </div>
    <p class="ms-1.5 text-stone-500">2</p>
  </label><label class="flex items-center select-none group leading-none flex-wrap disabled-utility"><input class="sr-only peer" type="radio" value="3">
    <div class="relative w-4 h-4 rounded-full border-2 focus-utility border-stone-500">
      <!--v-if-->
    </div>
    <p class="ms-1.5 text-stone-500">3</p>
  </label>
</fieldset>"
`;
